scalding-core/src/main/scala/com/twitter/scalding/DescribedExecutionApp.scala

```diff
@@ -0,0 +1,57 @@
+package com.twitter.scalding
+
+sealed abstract class ArgOptions
+
+object RequiredArg extends ArgOptions
+object OptionalArg extends ArgOptions
+object ListArg extends ArgOptions
```

Comment:
Good point, I wasn't a huge fan of exposing that parameter.  I also thought about the builder pattern but I liked the List because I can share a list of Described Args more easily between jobs.